pkg/controller/egressgateway/egressgateway_controller.go

```diff
@@ -423,6 +430,40 @@ func (r *ReconcileEgressGateway) reconcileEgressGateway(ctx context.Context, egw
 	return nil
 }
 
+func (r *ReconcileEgressGateway) getNetworkingPullSecrets(pullSecrets []*v1.Secret, ns string) ([]*v1.Secret, error) {
+	secrets := []*v1.Secret{}
+	for _, ps := range pullSecrets {
+		s := &v1.Secret{}
+		err := r.client.Get(context.Background(), client.ObjectKey{Name: ps.Name, Namespace: ns}, s)
+		if err != nil {
+			return nil, err
+		}
+		secrets = append(secrets, s)
+	}
+
+	return secrets, nil
+}
+
+func (r *ReconcileEgressGateway) patchFelixConfig(ctx context.Context) (*crdv1.FelixConfiguration, error) {
+	// Fetch any existing default FelixConfiguration object.
+	fc := &crdv1.FelixConfiguration{}
+	err := r.client.Get(ctx, types.NamespacedName{Name: "default"}, fc)
+
+	if err != nil && !errors.IsNotFound(err) {
+		r.status.SetDegraded(operatorv1.ResourceReadError, "Unable to read FelixConfiguration", err, log)
+		return nil, err
+	}
+
+	patchFrom := client.MergeFrom(fc.DeepCopy())
+	if fc.Spec.PolicySyncPathPrefix == "" {
+		fc.Spec.PolicySyncPathPrefix = "/var/run/nodeagent"
+	}
+	if err := r.client.Patch(ctx, fc, patchFrom); err != nil {
```

Comment:
We should only send the patch if we needed to change the felix configuration, if the value was already correct, we should skip the `Patch` call.